bluesky/callbacks/core.py

```diff
@@ -552,6 +372,142 @@ def event(self, doc):
         self.im.set_array(self._Idata)
 
 
+class LiveTable(CallbackBase):
+    '''Live updating table
+
+    Parameters
+    ----------
+    fields : list
+         List of fields to add to the table.
+
+    print_header_interval : int, optional
+         Reprint the header every this many lines, defaults to 50
+
+    min_width : int, optional
+         The minimum width is spaces of the data columns.  Defaults to 12
+
+    dflt_prec : int, optional
+         Precision to use if it can not be found in descriptor, defaults to 3
+
+    extra_pad : int, optional
+         Number of extra spaces to put around the printed data, defaults to 1
+
+    logbook : callable, optional
+        Must take a sting as the first positional argument
+
+           def logbook(input_str):
+                pass
+
+    '''
+    _FMTLOOKUP = {'s': '{pad}{{{k}: >{width}.{prec}{dtype}}}{pad}',
+                  'f': '{pad}{{{k}: >{width}.{prec}{dtype}}}{pad}',
+                  'g': '{pad}{{{k}: >{width}.{prec}{dtype}}}{pad}',
+                  'd': '{pad}{{{k}: >{width}{dtype}}}{pad}'}
+    _FMT_MAP = {'number': 'f',
+                'integer': 'd',
+                'string': 's',
+                }
+    _fm_sty = namedtuple('fm_sty', ['width', 'prec', 'dtype'])
+    water_mark = "{st[plan_type]} ['{st[uid]:.6s}'] (scan num: {st[scan_id]})"
+
+    def __init__(self, fields, *, print_header_interval=50,
+                 min_width=12, dflt_prec=3, extra_pad=1,
+                 logbook=None):
+        super().__init__()
+        self._header_interval = print_header_interval
+        # expand objects
+        self._fields = _get_obj_fields(fields)
+        self._start = None
+        self._stop = None
+        self._descriptors = set()
+        self._pad_len = extra_pad
+        self._extra_pad = ' ' * extra_pad
+        self._min_width = min_width
+        self._dflt_prec = dflt_prec
+        self._format_info = OrderedDict([
+            ('seq_num', self._fm_sty(10 + self._pad_len, '', 'd')),
+            ('time', self._fm_sty(10 + 2 * extra_pad, 10, 's'))
+        ])
+        self._rows = []
+        self.logbook = logbook
+
+    def descriptor(self, doc):
+        def patch_up_precision(p):
+            try:
+                return int(p)
+            except (TypeError, ValueError):
+                return self._dflt_prec
+
+        dk = doc['data_keys']
+        self._format_info.update(
+            ((k, self._fm_sty(
```

Comment:
A dictionary update using a tuple of every kv pair with a namedtuple using positional arguments and some unexplained math + magic numbers... 

I'm genuinely curious (and not in a sarcastic way). Do you seriously find the PR code to be more readable than something like this?
```python
for k in self._fields:
    width = max(self._min_width, 
                len(k) + 2, 
                self._dflt_prec + 1 + 2 * self._pad_len)
    dk_entry = dk[k]
    fmt = self._fm_sty(width=width,
                prec=patch_up_precision(dk_entry.get('precision', self._dflt_prec)),
                dtype=self._FMT_MAP[dk_entry['dtype']]))
             )
    self._format_info[k] = fmt
```